### PR TITLE
Added a new optional flag (-c) for the instance if you wanted the nam…

### PR DESCRIPTION
**Pull request by tiffanycmeyer13**
…e of the file on the nexus server to be different than the filename you were uploading

### CODE REVIEW

**Comment by tiffanycmeyer13** [comment]
Sean, you can decide if you want to incorporate this or add any further checks, but I thought I'd at least give you the option if you want to merge it in

**Comment by lesserwhirls** [comment]
The only that that jumps out to me is that we'd want to check that the array of filenames (`EXPRESSION`) is not greater than 1 when the new `-c` flag is used, since that combo does not make sense.

**Comment by tiffanycmeyer13** [comment]
Not sure what exit status I should put (not super familiar with bash...)

File: bash/nexus-upload.sh [hunk 1]
```diff
@@ -194,6 +210,10 @@ else
   EXPRESSION="$*"
 fi
 
+if [[ $# > 1 ]] && [[ ${NEW_FILENAME} != "" ]]; then
+  exit "Cannot use the change filename (-c flag) with multiple file uploads"
```
Comment (lesserwhirls) [inline bash/nexus-upload.sh:214]: Would you mind changing this to:

~~~bash
    echo "Cannot use the change filename (-c flag) with multiple file uploads"
    exit 1
~~~

File: bash/nexus-upload.sh [hunk 2]
```diff
@@ -91,6 +102,7 @@ Optional flags:
       echo upload command, but do not execute
   -f: use filename only
       do not preserve local path in the server side path
+  -c: change the filename on the server side
```
Comment (lesserwhirls) [inline bash/nexus-upload.sh:105]: Might be worth adding `(incompatible when uploading multiple files)`

**Review by lesserwhirls — CHANGES_REQUESTED** [review]

**Comment by lesserwhirls** [comment]
Awesome = thanks @tiffanycmeyer13!

**Comment by lesserwhirls** [comment]
Holding off on merging for now as we will pair up tomorrow to rebase and clean up the history.

**Review by lesserwhirls — APPROVED** [review]

**Pull request merged**